File: backend/src/main/java/com/campussync/campussync_backend/dto/LoginResponse.java
package com.campussync.campussync_backend.dto;

public class LoginResponse {

    private String token;
    private Long userId;
    private String name;
    private String email;
    private String role;
    private boolean firstLogin;

    public LoginResponse(
            String token,
            Long userId,
            String name,
            String email,
            String role,
        boolean firstLogin) {

        this.token = token;
        this.userId = userId;
        this.name = name;
        this.email = email;
        this.role = role;
        this.firstLogin=firstLogin;
    }

    public String getToken() {
        return token;
    }

    public Long getUserId() {
        return userId;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public String getRole() {
        return role;
    }
    public boolean isFirstLogin(){
        return firstLogin;
    }
}

File: frontend/lib/models/login_response.dart
class LoginResponse {
  final String token;
  final int userId;
  final String name;
  final String email;
  final String role;
  final bool firstLogin;

  const LoginResponse({
    required this.token,
    required this.userId,
    required this.name,
    required this.email,
    required this.role,
    required this.firstLogin,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      token: json['token'] as String,
      userId: json['userId'] as int,
      name: json['name'] as String,
      email: json['email'] as String,
      role: json['role'] as String,
      firstLogin: json['firstLogin'] as bool,
    );
  }
}